Src/Plugins/Security/SEC101_015.AkamaiCredentialsValidator.cs

```diff
@@ -14,15 +15,84 @@ namespace Microsoft.CodeAnalysis.Sarif.PatternMatcher.Plugins.Security
+            string authHeader = GetAuthorizationHeaderValue(secret, timestamp, authData, requestData);
+            request.Headers.Add("Authorization", authHeader);
+
+            return request;
+        }
+
+        internal static string GetRequestData(string method,
+                                              Uri uri)
         {
-            if (!groups.TryGetNonEmptyValue("id", out FlexMatch id) ||
-                !groups.TryGetNonEmptyValue("host", out FlexMatch host) ||
-                !groups.TryGetNonEmptyValue("secret", out FlexMatch secret) ||
-                !groups.TryGetNonEmptyValue("resource", out FlexMatch resource))
+            string headers = string.Empty;
+            string bodyHash = string.Empty;
+
+            return string.Format("{0}\t{1}\t{2}\t{3}\t{4}\t{5}\t",
+                method.ToUpper(),
+                uri.Scheme,
+                uri.Host,
+                uri.PathAndQuery,
+                headers,
+                bodyHash);
+        }
+
+        internal static string GetAuthDataValue(string id,
+                                                string resource,
+                                                string timestamp,
+                                                string scanIdentityGuid)
+        {
+            return string.Format("{0} client_token={1};access_token={2};timestamp={3};nonce={4};",
+                "EG1-HMAC-SHA256",
+                id,
+                resource,
+                timestamp,
+                scanIdentityGuid.ToLower());
+        }
+
+        internal static string GetAuthorizationHeaderValue(string secret,
+                                                           string timestamp,
+                                                           string authData,
+                                                           string requestData)
+        {
+            string hashType = "HMACSHA256";
+
+            byte[] time = Encoding.UTF8.GetBytes(timestamp);
+            string signingKey = Convert.ToBase64String(ComputeKeyedHash(time, secret, hashType));
+
+            byte[] data = Encoding.UTF8.GetBytes(string.Format("{0}{1}", requestData, authData));
+            string authSignature = Convert.ToBase64String(ComputeKeyedHash(data, signingKey, hashType));
+
+            return string.Format("{0}signature={1}", authData, authSignature);
+        }
+
+        internal static byte[] ComputeKeyedHash(byte[] data,
+                                                string key,
+                                                string hashType)
+        {
+            using (var algorithm = HMAC.Create(hashType.ToString()))
             {
-                return ValidationResult.CreateNoMatch();
+                algorithm.Key = Encoding.UTF8.GetBytes(key);
+                return algorithm.ComputeHash(data);
             }
+        }
+
+        protected override IEnumerable<ValidationResult> IsValidStaticHelper(IDictionary<string, FlexMatch> groups)
```

Comment:
this is a good max length. if any method is longer than this, we should break in multiple lines